src/macros.rs

```diff
@@ -261,7 +261,7 @@ rules.extend(vec![
 #[derive(Debug)]
 struct MySillyApplier(&'static str);
 impl Applier<SimpleLanguage, ()> for MySillyApplier {
-    fn apply_one(&self, _: &mut EGraph, _: Id, _: &Subst) -> Vec<Id> {
+    fn apply_one(&self, _: &mut EGraph, _: Id, _: &Subst) -> (Vec<Id>, Option<PatternAst<SimpleLanguage>>) {
```

Comment:
Hmm, what if different applications have different PatternAst? Same for searchers.

Reply:
Also, this is awkward. In most cases, I would rather the PatternAst be a reference so it doesn't have to copy it. But custom appliers have to build their own ast often, so it can't be a reference there.